code/modules/mob/living/say.dm

```diff
@@ -139,13 +139,49 @@ var/global/list/channel_to_radio_key = new
 	message_data[1] = message
 	message_data[2] = verb
 
+// Grabs any radios equipped to the mob, with message_mode used to
+// determine relevancy. See handle_message_mode below.
+/mob/living/proc/get_radios(var/message_mode)
+
+	if(!message_mode)
+		return
+
+	var/list/possible_radios
+	if(message_mode == "right ear" || message_mode == "left ear")
+		var/use_right = message_mode == "right ear"
+		var/obj/item/thing = get_equipped_item(use_right ? slot_r_ear_str : slot_l_ear_str)
+		if(thing)
+			LAZYDISTINCTADD(possible_radios, thing)
+		else
+			thing = get_equipped_item(use_right ? BP_R_HAND : BP_L_HAND)
+			if(thing)
+				LAZYDISTINCTADD(possible_radios, thing)
+	else
+		for(var/slot in global.ear_slots)
+			var/thing = get_equipped_item(slot)
+			if(thing)
+				LAZYDISTINCTADD(possible_radios, thing)
+
+	if(length(possible_radios))
+		for(var/atom/movable/thing as anything in possible_radios)
+			var/obj/item/radio/radio = thing.get_radio(message_mode)
+			if(istype(radio))
+				LAZYADD(., radio)
```

Comment:
You use LAZYDISTINCTADD earlier, which is slow, but only LAZYADD here. Is it not possible that get_radio on two of the distinct objects in `possible_radios` might return the same value somehow? This would have all the expense of using LAZYDISTINCTADD but lose the benefits by not doing it here.

Again, not blocking, just a comment.